wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extend.kt

```diff
@@ -38,9 +39,17 @@ class Extend private constructor(
     }
   }
 
-  fun validate(linker: Linker) {
+  fun validate(
+    linker: Linker,
+    syntaxRules: SyntaxRules
+  ) {
     val linker = linker.withContext(this)
     linker.validateImport(location, type!!)
+
+    if (syntaxRules.extensionIsPermittedOnlyForCustomOptions() &&
+        ProtoType.get(name) !in GOOGLE_PROTOBUF_OPTION_TYPES) {
+      linker.addError("extensions are permitted only for custom options [proto3]")
```

Comment:
Maybe simply `"extensions are not allowed [proto3]"`? I feel, same as the method name, that trying to convey we can actually extend on options makes thing awkward.

Reply:
Just copy-and-pasted from the Google proto document lol
I'm not familiar with the wire terminology, so I'd be happy to fix it!

Reply:
[memo]
SyntaxRule is not just a config, e.g. a list of booleans for a particular syntax rule. It can have some logic to tell the API user whether or not a specific syntax is met.

Reply:
[memo]
`Linker` doesn't have not know if the log is proto3 specific. It might be a good idea that `SyntaxRule` could have an API that generates some useful message to explain why the syntax error happened to the API user. This is more user-friendly and the user can figure out how to fix the error. It should be ok for SyntaxRule to know how to fix the error (since it concluded to fail the syntax for some reason).